ballerina/sql_client.bal

```diff
@@ -207,7 +207,12 @@ public client class SQLClient {
                 record {}[] arr = [];
                 error? e = from record {} item in joinStream
                     do {
-                        arr.push(check item.cloneWithType(joinMetadata.entity));
+                        record {}|error clonedItem = item.cloneWithType(joinMetadata.entity);
+                        if clonedItem is record {} {
+                            arr.push(clonedItem);
+                        } else {
+                            return <Error>clonedItem;
+                        }
```

Comment:
Can we add tests for the error case?

Reply:
Is there a specific need? Technically the type for the table would not be changed dueing code execution.

Reply:
Why do we need to do `cloneWithType` then?

Reply:
Looping in @kaneeldias

Reply:
`cloneWithType()` is not necessary. Will send a PR to fix